src/client/Microsoft.Identity.Client/Instance/CiamAuthority.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class CiamAuthority : AadAuthority
+    {
+        internal CiamAuthority(AuthorityInfo authorityInfo) : 
+            base(authorityInfo)
+        { }
+
+        /// <summary>
+        /// Translates CIAM authorities into a usable form. This is needed only until ESTS is updated to support the north star format
+        /// North star format: https://idgciamdemo.ciamlogin.com
+        /// Transformed format: https://idgciamdemo.ciamlogin.com/idgciamdemo.onmicrosoft.com
+        /// </summary>
+        internal static Uri TransformAuthority(Uri ciamAuthority)
+        {
+            string transformedInstance;
+            string transformedTenant;
+
+            string host = ciamAuthority.Host + ciamAuthority.AbsolutePath;
+            if (string.Equals(ciamAuthority.AbsolutePath, "/"))
+            {
+                string ciamTenant = host.Substring(0, host.IndexOf(Constants.CiamAuthorityHostSuffix, StringComparison.OrdinalIgnoreCase));
+                transformedInstance = $"https://{ciamTenant}{Constants.CiamAuthorityHostSuffix}/";
+                transformedTenant = ciamTenant + ".onmicrosoft.com";
+                return new Uri(transformedInstance + transformedTenant);
```

Comment:
Tried to remove the Uri parsing as much as I can but the logic ends up becoming a bit more complex if I remove it from everywhere. I did reduce the number of parses in the hotpash so there should still be a performance improvement overall.